protocol/protocol.go

```diff
@@ -143,7 +145,15 @@ func NewChain(ctx context.Context, store Store, pool Pool, heights <-chan uint64
 		return nil, errors.Wrap(err, "looking up blockchain height")
 	}
 
-	// Note that c.height.n may still be zero here.
+	if c.state.height >= 1 {
+		b, err := store.GetBlock(ctx, 1)
+		if err != nil {
+			return nil, errors.Wrap(err, "retrieving initial block")
+		}
+		c.InitialBlockHash = b.Hash()
+	}
+
+	// Note that c.state.height may still be zero here.
```

Comment:
If `c.state.height` is 0, will `c.InitialBlockHash` get set on the first block committed?